File: C++/check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        if (s1 == s2) {
            return true;
        } else {
            int sz = s1.size();
            int pos = -1;
            char c1 = 0, c2 = 0;
            for (int i = 0; i < sz; i++) {
                if (s1[i] != s2[i]) {
                    if (pos == -1) {
                        pos = i;
                        c1 = s1[i];
                        c2 = s2[i];
                    } else if (s2[i] == c1 && s1[i] == c2) {
                        c1 = 0;
                        c2 = 0;
                    } else {
                        return false;
                    }
                }
            }
            return c1 == 0;
        }
    }
};

File: C#/check-if-one-string-swap-can-make-strings-equal.cs
public class Solution {
    public bool AreAlmostEqual(string s1, string s2) {
        if (s1.Length != s2.Length) {
            return false;
        }
        int a = -1, b = -1;
        for (int i = 0; i < s1.Length; i++) {
            if (s1[i] != s2[i]) {
                if (a == -1) {
                    a = i;
                } else if (b == -1) {
                    b = i;
                } else {
                    return false;
                }
            }
        }
        if (a == -1) {
            return true;
        }
        if (b == -1) {
            return false;
        }
        return s1[a] == s2[b] && s1[b] == s2[a];
    }
}